### PR TITLE
Changes to the constructors, additional getter and setter and minor fixes

### PR DESCRIPTION
**Pull request by W1therhunter**
I modified some of the Constructors of the Component-Subclasses. A few parameters where missing, so with the modifications is now possible to assign the values to the attributes. Before it was only possible by using Setters after object-creation.
Speaking of Setters for the Memory Component there is the attribute "is_volatile" and now the attributes value can be assigned via the constructor or with the respective setter.
Additionally there were some errors occurring while compiling that are now fixed by minor adjustments.

### CODE REVIEW

**Comment by stepanvanecek** [comment]
also, pelase target your PR towards develop, not master

File: src/Component.cpp [hunk 1]
```diff
@@ -650,6 +650,8 @@ void Numa::SetSize(long long _size) { size = _size;}
 
 long long Memory::GetSize() {return size;}
 void Memory::SetSize(long long _size) {size = _size;}
+bool Memory::IsVolatile() {return is_volatile;}
```
[the remark this answers is not in the document]
Comment (W1therhunter) [reply src/Component.cpp:653]: "volatile" seems to be reserved by c++. Could we give it an alternative name?

[the remark this answers is not in the document]
Comment (stepanvanecek) [reply src/Component.cpp:653]: ah, true, probably that's why we have this name.. then leave the name is_volatile and just name the getter GetIsVolatile

File: src/Component.cpp [hunk 2]
```diff
@@ -699,25 +701,27 @@ Topology::Topology():Component(0, "sys-sage Topology", SYS_SAGE_COMPONENT_TOPOLO
...
 
 Cache::Cache(int _id, int  _cache_level, long long _cache_size, int _associativity, int _cache_line_size): Component(_id, "Cache", SYS_SAGE_COMPONENT_CACHE), cache_type(to_string(_cache_level)), cache_size(_cache_size), cache_associativity_ways(_associativity), cache_line_size(_cache_line_size){}
 Cache::Cache(Component * parent, int _id, string _cache_type, long long _cache_size, int _associativity, int _cache_line_size): Component(parent, _id, "Cache", SYS_SAGE_COMPONENT_CACHE), cache_type(_cache_type), cache_size(_cache_size), cache_associativity_ways(_associativity), cache_line_size(_cache_line_size){}
 Cache::Cache(Component * parent, int _id, int _cache_level, long long _cache_size, int _associativity, int _cache_line_size): Cache(parent, _id, to_string(_cache_level), _cache_size, _associativity, -1){}
 
-Subdivision::Subdivision(Component * parent, int _id, string _name, int _componentType): Component(parent, _id, _name, _componentType)
+Subdivision::Subdivision(Component * parent, int _id, string _name, int _componentType, int _subdivisionType): Component(parent, _id, _name, _componentType), type(_subdivisionType)
```
[the remark this answers is not in the document]
Comment (W1therhunter) [reply src/Component.cpp:719]: Alright, I thought it should be included because we have a public Setter for this variable too.

**Comment by stepanvanecek** [comment]
Closing as another PR to develop was opened

**Pull request closed without merging**